### ISSUE TITLE
[Bug]: QuickSight templates are breaking CDKTF builds

### ISSUE THREAD

**Issue by yuriy-yarosh** [hashicorp/terraform-provider-aws#31615]
### Terraform Core Version

1.4.6

### AWS Provider Version

5.0.1

### Affected Resource(s)

aws_appsync_domain_name
aws_quicksight_template

### Expected Behavior

CDKTF should be able to get AWS provider 

### Actual Behavior

JSII generates 66 index structs with conflicts for the Quicksight Template resource Interface names, and the AppSync Domain Name.

### Relevant Error/Panic Output Snippet

```shell
yarn run tf_import
yarn run v1.22.19
$ cd platform/utility/tfmodules && node --max-semi-space-size=512 --max-old-space-size=1536 --optimize-for-size --gc-interval=500 ../../../node_modules/.bin/cdktf get -l go -o imports --parallelism 4
NOTE: Temp directory retained due to an error: /tmp/temp-GgXm2Z
Error: jsii compilation failed with non-zero exit code: 1
  | [2023-05-28T04:47:16.245] [ERROR] jsii/compiler - Type model errors prevented the JSII assembly from being created
  | warning JSII6: A "peerDependency" on "constructs" at "10.2.34" means you should take a "devDependency" on "constructs" at "10.2.34" (found "undefined")
  | warning JSII6: A "peerDependency" on "cdktf" at "0.16.1" means you should take a "devDependency" on "cdktf" at "0.16.1" (found "undefined")
  | warning JSII3: There is no "README.md" file. It is required in order to generate valid PyPI (Python) packages.
  | node_modules/cdktf/lib/complex-computed-list.d.ts:81:1 - error JSII3005: Type "cdktf.ComplexList" cannot be used as an interface
  |  81 export declare abstract class ComplexList extends ComplexResolvable implements ITerraformAddressable {
  |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  |  82     protected terraformResource: IInterpolatingParent;
  |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | ...
  |  86     computeFqn(): string;
  |     ~~~~~~~~~~~~~~~~~~~~~~~~~
  |  87 }
  |     ~
  | providers/aws/appsync-domain-name/index.ts:79:14 - warning JSII5019: The property name "appsyncDomainName" conflicts with the declaring class "AppsyncDomainName". This will result in renaming the class to "_AppsyncDomainName" in C#. Consider renaming "appsyncDomainName".
  | 79   public get appsyncDomainName() {
  |                 ~~~~~~~~~~~~~~~~~
  |   providers/aws/appsync-domain-name/index.ts:34:14
  |     34 export class AppsyncDomainName extends cdktf.TerraformResource {
  |                     ~~~~~~~~~~~~~~~~~
  |     The declaring class is introduced here
  | providers/aws/quicksight-template/index-structs/structs400.ts:9180:14 - error JSII8007: Interface contains behavior. Rename "QuicksightTemplateDefinitionSheetsFilterControlsList" to "IQuicksightTemplateDefinitionSheetsFilterControlsList"
  | 9180 export class QuicksightTemplateDefinitionSheetsFilterControlsList extends cdktf.ComplexList {
  |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/aws/quicksight-template/index-structs/structs800.ts:7363:14 - error JSII8007: Interface contains behavior. Rename "QuicksightTemplateDefinitionSheetsParameterControlsList" to "IQuicksightTemplateDefinitionSheetsParameterControlsList"
  | 7363 export class QuicksightTemplateDefinitionSheetsParameterControlsList extends cdktf.ComplexList {
  |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  +----------------------------------------------------------------------------------
  | Command: node_modules/jsii/bin/jsii --silence-warnings reserved-word
  | Workdir: /tmp/temp-GgXm2Z
  +----------------------------------------------------------------------------------
    at p (node_modules/cdktf-cli/bundle/bin/cmds/handlers.js:51:1581)
    at ChildProcess.<anonymous> (node_modules/cdktf-cli/bundle/bin/cmds/handlers.js:54:112)
    at Object.onceWrapper (node:events:628:26)
    at ChildProcess.emit (node:events:513:28)
    at ChildProcess.emit (node:domain:489:12)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12)
⠼ downloading and generating modules and providers...
Error: jsii compilation failed with non-zero exit code: 1
  | [2023-05-28T04:47:16.245] [ERROR] jsii/compiler - Type model errors prevented the JSII assembly from being created
  | warning JSII6: A "peerDependency" on "constructs" at "10.2.34" means you should take a "devDependency" on "constructs" at "10.2.34" (found "undefined")
  | warning JSII6: A "peerDependency" on "cdktf" at "0.16.1" means you should take a "devDependency" on "cdktf" at "0.16.1" (found "undefined")
  | warning JSII3: There is no "README.md" file. It is required in order to generate valid PyPI (Python) packages.
  | node_modules/cdktf/lib/complex-computed-list.d.ts:81:1 - error JSII3005: Type "cdktf.ComplexList" cannot be used as an interface
  |  81 export declare abstract class ComplexList extends ComplexResolvable implements ITerraformAddressable {
  |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  |  82     protected terraformResource: IInterpolatingParent;
  |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | ...
  |  86     computeFqn(): string;
  |     ~~~~~~~~~~~~~~~~~~~~~~~~~
  |  87 }
  |     ~
  | providers/aws/appsync-domain-name/index.ts:79:14 - warning JSII5019: The property name "appsyncDomainName" conflicts with the declaring class "AppsyncDomainName". This will result in renaming the class to "_AppsyncDomainName" in C#. Consider renaming "appsyncDomainName".
  | 79   public get appsyncDomainName() {
  |                 ~~~~~~~~~~~~~~~~~
  |   providers/aws/appsync-domain-name/index.ts:34:14
  |     34 export class AppsyncDomainName extends cdktf.TerraformResource {
  |                     ~~~~~~~~~~~~~~~~~
  |     The declaring class is introduced here
  | providers/aws/quicksight-template/index-structs/structs400.ts:9180:14 - error JSII8007: Interface contains behavior. Rename "QuicksightTemplateDefinitionSheetsFilterControlsList" to "IQuicksightTemplateDefinitionSheetsFilterControlsList"
  | 9180 export class QuicksightTemplateDefinitionSheetsFilterControlsList extends cdktf.ComplexList {
  |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/aws/quicksight-template/index-structs/structs800.ts:7363:14 - error JSII8007: Interface contains behavior. Rename "QuicksightTemplateDefinitionSheetsParameterControlsList" to "IQuicksightTemplateDefinitionSheetsParameterControlsList"
  | 7363 export class QuicksightTemplateDefinitionSheetsParameterControlsList extends cdktf.ComplexList {
  |                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  +----------------------------------------------------------------------------------
  | Command: node_modules/jsii/bin/jsii --silence-warnings reserved-word
  | Workdir: /tmp/temp-GgXm2Z
  +----------------------------------------------------------------------------------
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Terraform Configuration Files

None

### Steps to Reproduce

Both cdktf import for either AWS 5.0.1 or 4.67.0 provider versions do fail.
As per [CDKTF #2879](https://github.com/hashicorp/terraform-cdk/issues/2879).

Last Confirmed working AWS provider version is **4.66.1** 

### Would you like to implement a fix?

No

**Comment by yuriy-yarosh** [comment]
Use node18 LTS to reproduce, with

`package.json`
```json
{
  "name": "cdktf-collapses",
  "description": "bench",
  "license": "CLOSED",
  "version": "0.1.1",
  "main": "",
  "types": "",
  "jsii": {},
  "engines": {
    "node": "18"
  },
  "scripts": {
    "import": "node --max-semi-space-size=512 --max-old-space-size=1536 --optimize-for-size --gc-interval=500 ../../../node_modules/.bin/cdktf get -l go -o imports --parallelism 4",
  },
  "devDependencies": {
    "cdktf": "~0.16.1",
    "cdktf-cli": "~0.16.1",
    "constructs": "~10.2.35",
    "jsii": "~5.0.7"
  }
}
```

`cdktf.json`
```json
{
  "language": "golang",
  "app": "echo Doesn't import",
  "terraformProviders": [
    "aws@~> 5.0.1"
  ]
}
```

And import with `npm run import`.

Additional `--optimize-for-size --gc-interval=500` help with mem consumption, but can't be passed via NODE_OPTIONS anymore, so it's possible to restrict it to fit into 1500 old space.

**Comment by justinretzolk** [comment]
Hey @yuriy-yarosh 👋 Thank you for taking the time to raise this! I noticed that the issue raised in the terraform-cdk repository has since been resolved. Can you confirm whether this is still occurring?

**Comment by yuriy-yarosh** [comment]
@justinretzolk it should've been resolved in [terraform-cdk#2895](https://github.com/hashicorp/terraform-cdk/pull/2895).
But I have no means to confirm it anymore, feel free to close this issue after confirming that it's been resolved.
